python/social_bot/gazebo_agent.py

```diff
@@ -170,6 +190,25 @@ def get_observation(self, teacher, sentence_raw="hello"):
             obs = self.get_camera_observation()
         else:  # observation is pure low-dimentional states
             obs = teacher.get_task_specific_observation(self)
+            if self._goal_conditioned:
+                flat_obs = obs
+                obs = OrderedDict()
+                obs['observation'] = flat_obs
+                obs['achieved_goal'] = flat_obs[6:8]
+                obs['desired_goal'] = flat_obs[12:14]
+                if self._use_aux_achieved:
+                    obs['observation'] = flat_obs[14:]
+                    obs['aux_achieved'] = np.concatenate(
+                        (flat_obs[:6], flat_obs[8:12]), axis=0)
+                    if self._xy_only_aux:
+                        # 2: z-speed, 3, 4: angular velocity, 5: yaw-vel, 6, 7: x, y, 8: z, 9, 10, 11: pose
+                        obs['observation'] = np.concatenate(
+                            (flat_obs[2:5], flat_obs[8:11], flat_obs[14:]),
```

Comment:
Agree with @Jialn that should keep GazeboBase clean. Specialized logic for particular task should be in other places.

Reply:
Done.